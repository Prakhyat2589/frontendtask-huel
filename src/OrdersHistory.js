import React, { useState, useEffect } from 'react';
import axios from 'axios';
import noimage from './assests/images/noimage.png';

const OrdersHistory = () => {
  const [orderdata, setOrderData] = useState();

  const fetchOrdersHistory = async () => {
    try {
      const fetchURL = await axios.get(
        'https://frontendtest.huel.io/api/line-items'
      );
      const response = fetchURL.data.line_items;
      setOrderData(response);
    } catch (e) {
      console.log('Network Error');
    }
  };

  //   const checkImage = async image_url => {
  //     var http = new XMLHttpRequest();

  //     http.open('HEAD', image_url, false);
  //     http.send();

  //     return http.status != 404;
  //   };

  //   const formatter = new Intl.NumberFormat('en-US', {
  //     style: 'currency',
  //     currency: 'USD'
  //   });

  useEffect(() => {
    fetchOrdersHistory();
    // console.log(checkImage);

    // console.log(formatter.format(1000));
  }, []);

  return (
    <div className="order-information-expanded">
      <div className="product-list-boxes columns is-multiline">
        {orderdata &&
          orderdata.map((item, index) => {
            return (
              <div className="column is-6" key={index} data-testid="orders">
                <div className="media">
                  <div className="media-left">
                    <img alt={item.title} className="image" src={item.image} />
                  </div>
                  <div className="media-content">
                    <div>
                      <p className="product-title">{item.title}</p>
                      <p className="product-variants">
                        {item.quantity}x {item.name}
                      </p>
                    </div>
                  </div>
                  <div className="media-right">
                    <p className="product-price">${item.price}</p>
                  </div>
                </div>
              </div>
            );
          })}
      </div>
    </div>
  );
};

export default OrdersHistory;
