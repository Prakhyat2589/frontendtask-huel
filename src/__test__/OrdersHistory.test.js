import React from 'react';
import { render, cleanup, waitFor } from '@testing-library/react';
import OrdersHistory from '../OrdersHistory';
import axios from 'axios';

jest.mock('axios');

afterEach(cleanup);
describe('OrdersHistory component render', () => {
  it('fetches and displays data', async () => {
    // We'll be explicit about what data Axios is to return when `get` is called.
    axios.get.mockResolvedValueOnce({
      data: {
        line_items: [
          {
            title: 'Huel Shaker Bottle (Clear)'
          },
          {
            title: 'Free T-Shirt & Shaker'
          }
        ]
      }
    });
    const url = 'https://frontendtest.huel.io/api/line-items';
    const { getByText, getAllByTestId } = render(<OrdersHistory />);

    await waitFor(() =>
      getAllByTestId('orders').map(orders => orders.textContent)
    );
    expect(getByText('Huel Shaker Bottle (Clear)'));
    expect(axios.get).toHaveBeenCalledTimes(1);
    expect(axios.get).toHaveBeenCalledWith(url);
  });
  it('fetches no data from an API', async () => {
    const message = 'Network Error';
    axios.get.mockResolvedValue(message);
    const { component } = render(<OrdersHistory />);
    expect(component).toMatchSnapshot();
  });
});
