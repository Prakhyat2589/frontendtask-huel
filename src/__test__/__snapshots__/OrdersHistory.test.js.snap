// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrdersHistory component render fetches no data from an API 1`] = `undefined`;
